feat: add `disable_limit_max_threads` feature and limit the max threads to 2 by default for CRAN compatibility (#720)
pola-rs/r-polars: NEWS.md

### What's changed

> -   For compatibility with CRAN, the number of threads used by Polars may be automatically set to 2 (#720).

Reviewer: ```suggestion
-   For compatibility with CRAN, the number of threads used by Polars is automatically set to 2 (#720).
```

    To disable this behavior and have the maximum number of threads used automatically,
    one of the following ways can be used:
    -   Build the Rust library with the `disable_auto_limit_max_threads` feature.

>     -   Set the `polars.disable_auto_limit_max_threads` option to `TRUE` with the `options()` function

Reviewer: I find it hard to understand what `disable_auto_limit_max_threads = TRUE` does, can we rename if `use_max_threads`? It's clearer to set `polars.use_max_threads = TRUE`

Author: Good point.
I was wondering if it would be worth aligning the name with feature, but it would definitely be easier to understand if the option name was changed.

Author: I changed it to `polars.limit_max_threads`.
Given that the environment variable is `POLARS_MAX_THREADS`, I think the term `max` confusing.

>         before loading the package.

Reviewer: We could mention `Sys.setenv(POLARS_MAX_THREADS =)` so that users know that they don't have only the choice of 2 threads or all of them, no?

Author: How about 87d5eebaa8ba8e04cedab5827c6c3277bdc006fc?
